build.sbt

```diff
@@ -15,7 +15,6 @@ libraryDependencies ++= Seq(
 
 // scripted test settings
 scriptedLaunchOpts += "-Dproject.version=" + version.value
-scriptedLaunchOpts += "-Dcodegen.samples.dir=" + ((baseDirectory in ThisBuild).value / "src/test/resources")
```

Comment:
I isolated the test cases as it was hard to refactor things due to the inter-dependencies.
Also the scripted tests acted as unit tests as well, which is too much IMHO.